spec/api/v1/team_invites_controller_spec.rb

```diff
@@ -1,5 +1,89 @@
 require 'rails_helper'
 
 RSpec.describe "Team Invites API", type: :api do
-  it "needs some tests"
+  let!(:user) { create(:user_with_team, email: "owner@openhq.io") }
+  let!(:project) { create(:project, name: "Test Project", team: user.teams.first) }
+  let!(:other_project) { create(:project, name: "Other Project", team: user.teams.first) }
+
+  before do
+    project.users << user
+  end
+
+  describe "POST /api/v1/team_invites" do
+    it "requires a valid email address" do
+      post "/api/v1/team_invites", { user: { email: "invite" } }, api_token_header(user)
+      expect(last_response.status).to eq(422)
+    end
+
+    it "invites a new user to the team" do
+      post "/api/v1/team_invites", { user: { email: "invite@openhq.io" } }, api_token_header(user)
+      expect(last_response.status).to eq(200)
+      expect(response_json[:team_invites].count).to eq(1)
+      expect(response_json[:team_invites].first[:team_id]).to eq(user.teams.first.id)
+      expect(response_json[:team_invites].first[:role]).to eq("user")
+      expect(response_json[:team_invites].first[:status]).to eq("invited")
+      expect(response_json[:team_invites].first[:invitation_code]).not_to be_empty
+      expect(response_json[:team_invites].first[:invited_at]).not_to be_empty
+      expect(response_json[:team_invites].first[:invited_by]).to eq(user.id)
+    end
+
+    it "adds projects to the invited user" do
+      post "/api/v1/team_invites", { user: { email: "invite@openhq.io", project_ids: [project.id] } }, api_token_header(user)
+      expect(response_json[:projects].first[:name]).to eq("Test Project")
+    end
+
+    it "does not add projects the current user is not part of" do
+      post "/api/v1/team_invites", { user: { email: "invite@openhq.io", project_ids: [project.id, other_project.id] } }, api_token_header(user)
+      expect(response_json[:projects].count).to eq(1)
+      expect(response_json[:projects].first[:name]).to eq("Test Project")
+    end
+
+    it "will not invite a user if they are already on the team" do
+      post "/api/v1/team_invites", { user: { email: user.id } }, api_token_header(user)
+      expect(response_json[:team_invites]).to be_nil
```

Comment:
I don't like how this returns nil when its empty - cant figure out why it isn't returning an empty array